.github/workflows/build_and_test.yml

```diff
@@ -46,6 +46,8 @@ jobs:
         run: |
           fingerprint=$RANDOM
           echo "FINGERPRINT=$fingerprint" >> $GITHUB_ENV
+          SGX_MODE_LOWERCASE=$(echo "${${{ matrix.sgx_mode }},,}")
+          echo "IMAGE_SUFFIX=$SGX_MODE_LOWERCASE-${{ matrix.flavor_id }}-${{ github.sha }}" >> $GITHUB_ENV
```

Comment:
Introduce the `IMAGE_SUFFIX` environment variable to make the naming less verbose, and make it less error-prone to human errors. The image suffix does now also contain the sgx_mode: SW/HW.